tests/extension/test_manager.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+from jupyter_server.extension.manager import (
+    ExtensionPoint,
+    ExtensionPackage,
+    ExtensionManager,
+    ExtensionMetadataError,
+    ExtensionModuleNotFound
+)
+
+# Use ServerApps environment because it monkeypatches
+# jupyter_core.paths and provides a config directory
+# that's not cross contaminating the user config directory.
+pytestmark = pytest.mark.usefixtures("environ")
+
+
+def test_extension_point_api():
+    # Import mock extension metadata
+    from .mockextensions import _jupyter_server_extension_paths
+
+    # Testing the first path (which is an extension app).
+    metadata_list = _jupyter_server_extension_paths()
+    point = metadata_list[0]
+
+    module = point["module"]
+    app = point["app"]
+
+    e = ExtensionPoint(metadata=point)
+    assert e.module_name == module
+    assert e.name == app.name
+    assert app is not None
+    assert callable(e.load)
+    assert callable(e.link)
+
+
+def test_extension_point_metadata_error():
+    # Missing the "module" key.
+    bad_metadata = {"name": "nonexistent"}
+    with pytest.raises(ExtensionMetadataError):
+        ExtensionPoint(metadata=bad_metadata)
+
+
+def test_extension_point_notfound_error():
+    bad_metadata = {"module": "nonexistent"}
+    with pytest.raises(ExtensionModuleNotFound):
+        ExtensionPoint(metadata=bad_metadata)
+
+
+def test_extension_package_api():
+    # Import mock extension metadata
+    from .mockextensions import _jupyter_server_extension_paths
+
+    # Testing the first path (which is an extension app).
+    metadata_list = _jupyter_server_extension_paths()
```

Comment:
Extension points?  Or did you decide against the rename?

Reply:
No, you're totally right. I've updated with the new name. I still catch the old function name but a throw a debug message that this might change in the future.